### PR TITLE
Adding new packages to the Fit Model Keyboard. (Model>Fit Model> Fit Model Keyboard)

### LINKED ISSUE #7681: Add a Bayesian keyboard to the set in the Fit Models keyboard.
This is a suggestion for @Tula1 to undertake, perhaps together with @EstherNjeriLiberatta?  Also there should (soon after) also be a Bayesian keyboard (using the same packages) in the Use Models keyboard.  Perhaps @Tula1 can specify that as this issue is resolved.  Each package below will have some functions that start with data and others that use the model.

a) We have suggested that the `bayesglm` and `bayespolr` be added to the general fit model dialogue.  They should also be included here.  
b) The statsr package has the bayes_inference function and also the `inference` function. 

c) So there will be multiple packages and the new Help button with the down arrow, should allow access to each.  Also it is convenient to include the relevant functions in each package, even though the inference function is not Bayesian.  So maybe the entry is Bayesian Plus.  And the keys in b) could perhaps be called `Bayes_inf` and `inference` and be next to each other.

d) I think there will be more to add, but that will be a good start

### PR DESCRIPTION
**Pull request by Tula1**
Fixes #7681
In this PR, @EstherNjeriLiberatta and I have added the `bayesglm` and `bayespolr` functions from the `arm` package, and the `bayes_inference` and `inference` packages from the `statsr` package into the fit model keyboard. 
@rdstern, this is ready for review.

### CODE REVIEW

**Comment by N-thony** [comment]
@Tula1 which issue this PR attempts to fix?

**Comment by Tula1** [comment]
@N-thony , We are not fixing any issue, just adding Bayesian functions to the Fit Model Keyboard.

**Comment by rdstern** [comment]
@Tula1 I have added an issue #7870 that corresponds to your work.  Possibly with @EstherNjeriLiberatta you could link this pull request to Fix, or partially fix this issue.
When you are ready for the pull request to be tested, could you mention how you have done initial tests, so we know where to start.  In particular you could suggest appropriate data and then roughly what you did for your testing.
You added 2 keyboards, one for arm and the other for statsr.
a) The arm keyboard seems to work fine.  The link to the help also works.  This is great.
b) But I couldn't get the statsr keyboard functions to work.  If you decide to keep that keyboard, then I suggest it is wrongly placed anyway.  There is also a Hypothesis testing keyboard and I suggest it goes better there.

But you have to help the user more in those keys.  When you press a key it should provide sufficient that the user can fit the model by just adding a column or two, etc.  Here I had first to type the name of the data frame, which is compulsory.  That could be provided automatically.  I also had to complete various other arguments - in fact so many - and some not so obvious that I never got it to work.  Where there are many **necessary** options I suggest the pressing of the key automatically gives one of them.  So it will easily run for at least one example.

c) But as you are now into adding keyboards - and seem to do it well, could you look to swap (at least initially) the statsr keyboard for one on rstanarm?

**Comment by N-thony** [comment]
@Tula1 can you update about this?

**Comment by N-thony** [comment]
@anastasia-mbithe can you peer review this?

File: instat/dlgExtremesClimatic.vb [hunk 1]
```diff
@@ -52,6 +52,8 @@ Public Class dlgExtremesClimatic
     Private clsDeclusteringFunction As New RFunction
     Private clsDummyRfunction As New RFunction
```
Comment (anastasia-mbithe) [inline instat/dlgExtremesClimatic.vb:54]: Delete line 54, we don't need the empty line there.

File: instat/dlgExtremesClimatic.vb [hunk 2]
```diff
@@ -253,6 +255,15 @@ Public Class dlgExtremesClimatic
...
         ucrNudRunLength.SetMinMax(iNewMin:=1, iNewMax:=Integer.MaxValue)
 
+        ucrChkVariableName.SetText("NewVariable Name")
+        ucrChkVariableName.AddToLinkedControls(ucrSaveVariableName, {True}, bNewLinkedHideIfParameterMissing:=True, bNewLinkedAddRemoveParameter:=True)
+        ucrSaveVariableName.SetSaveTypeAsColumn()
+        ucrSaveVariableName.SetIsComboBox()
+        ucrSaveVariableName.SetLabelText("New Variable Name")
+        ucrSaveVariableName.SetDataFrameSelector(ucrSelectorClimaticExtremes.ucrAvailableDataFrames)
+        ucrSaveVariableName.SetPrefix("Clusters")
+
```
Comment (anastasia-mbithe) [inline instat/dlgExtremesClimatic.vb:265]: Also here

File: instat/dlgHypothesisTestsCalculator.vb [hunk 3]
```diff
@@ -1038,4 +1070,18 @@ Public Class dlgHypothesisTestsCalculator
...
+    End Sub
+
+    Private Sub SaveResults()
+        If ucrInputComboRPackage.GetText = "statsr" Then
+            ucrBase.clsRsyntax.RemoveAssignTo()
+        Else
+            ucrBase.clsRsyntax.SetAssignTo("Last_Test", strTempModel:="Last_Test", strTempDataframe:=ucrSelectorColumn.ucrAvailableDataFrames.cboAvailableDataFrames.SelectedItem)
+
+        End If
+
```
Comment (anastasia-mbithe) [inline instat/dlgHypothesisTestsCalculator.vb:1085]: delete also line 1083 and 1085

File: instat/dlgModelling.vb [hunk 4]
```diff
@@ -146,6 +149,25 @@ Public Class dlgModelling
...
+        clsStanGlmFunction.SetPackageName("rstanarm")
+        clsStanGlmFunction.SetRCommand("StanGlm")
+        clsStanGlmFunction.iCallType = 2
+
+        clsStanPolrFunction.SetPackageName("rstanarm")
+        clsStanPolrFunction.SetRCommand("StanPolr")
+        clsStanPolrFunction.iCallType = 2
+
+
+
```
Comment (anastasia-mbithe) [inline instat/dlgModelling.vb:170]: A single-line space will do

**Review by anastasia-mbithe — CHANGES_REQUESTED** [review]
@Tula1, change also the cursor position of both the inference function and bayes_inf function when we have Include arguments checked.

@N-thony, I have not testing how it works. It bugs on my side (I think its because of the dataset). I dont have the tapwater dataset. Maybe @rdstern could test that.

**Comment by rdstern** [comment]
@derekagorhom could you please have a look at this pull request, which currently has build errors.  It adds more keyboards to the Fit Model dialogue and I am sure your new R-Help stuff will also be needed there.

**Pull request closed without merging**